# Explanation to question number 7 in assignment- 1

In [24]:
import pandas as pd
import numpy as np

# loadong csv file using pandas

In [25]:
df = pd.read_csv('Q7.csv')

In [26]:
df

Unnamed: 0  Points  Score  Weigh
0             Mazda RX4    3.90  2.620  16.46
1         Mazda RX4 Wag    3.90  2.875  17.02
2            Datsun 710    3.85  2.320  18.61
3        Hornet 4 Drive    3.08  3.215  19.44
4     Hornet Sportabout    3.15  3.440  17.02
5               Valiant    2.76  3.460  20.22
6            Duster 360    3.21  3.570  15.84
7             Merc 240D    3.69  3.190  20.00
8              Merc 230    3.92  3.150  22.90
9              Merc 280    3.92  3.440  18.30
10            Merc 280C    3.92  3.440  18.90
11           Merc 450SE    3.07  4.070  17.40
12           Merc 450SL    3.07  3.730  17.60
13          Merc 450SLC    3.07  3.780  18.00
14   Cadillac Fleetwood    2.93  5.250  17.98
15  Lincoln Continental    3.00  5.424  17.82
16    Chrysler Imperial    3.23  5.345  17.42
17             Fiat 128    4.08  2.200  19.47
18          Honda Civic    4.93  1.615  18.52
19       Toyota Corolla    4.22  1.835  19.90
20        Toyota Corona    3.70  2.465  20.01
21     Dodge Challenger    2.76  3.520  16.87
22          AMC Javelin    3.15  3.435  17.30
23           Camaro Z28    3.73  3.840  15.41
24     Pontiac Firebird    3.08  3.845  17.05
25            Fiat X1-9    4.08  1.935  18.90
26        Porsche 914-2    4.43  2.140  16.70
27         Lotus Europa    3.77  1.513  16.90
28       Ford Pantera L    4.22  3.170  14.50
29         Ferrari Dino    3.62  2.770  15.50
30        Maserati Bora    3.54  3.570  14.60
31           Volvo 142E    4.11  2.780  18.60

# calculatng the mean, median, mode, Variance, Standard Deviation, and Range of a data frame using different statistical methods of python

# Calculating the mean median and standard deviation by using describe method

In [27]:
df.describe().round(2)

Points  Score  Weigh
count   32.00  32.00  32.00
mean     3.60   3.22  17.85
std      0.53   0.98   1.79
min      2.76   1.51  14.50
25%      3.08   2.58  16.89
50%      3.70   3.32  17.71
75%      3.92   3.61  18.90
max      4.93   5.42  22.90

# calculating the range

In [28]:
# RANGE = DIFFERENCE BETWEEN MIN AND MAX VALUE SO THE RANGE OF Points = 4.93-2.76 = 2.17 and RANGE OF Score = 5.42-1.51 = 3.91
# and the RANGE of Weigh = 22.90-14.50 =8.40

# Analysis

# for the given data the mean and median are almost simlar and by looking at standard deviaton the vales are not far from the mean.

# calculating the mode

In [29]:
df[['Points','Score','Weigh']].mode().round(2).count()

Points    2
Score     1
Weigh     2
dtype: int64

In [30]:
df[['Points','Score','Weigh']].mode().round(2)

Points  Score  Weigh
0    3.07   3.44  17.02
1    3.92    NaN  18.90

# Analysis of Mode

# in points 3.07 and 3.92 repeats more times and for score 3.44 repeats more times and for weigh 17.02 and 18.90 repets more times

# calculating the variance

In [31]:
df.var().round(2)

C:\Users\nishitha\AppData\Local\Temp\ipykernel_10924\2090591605.py:1: FutureWarning: The default value of numeric_only in DataFrame.var is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.var().round(2)


Points    0.29
Score     0.96
Weigh     3.19
dtype: float64

# Analysis of variance

# By looking at varinace , the vales are not spread far from their mean

In [32]:
df.head()

Unnamed: 0  Points  Score  Weigh
0          Mazda RX4    3.90  2.620  16.46
1      Mazda RX4 Wag    3.90  2.875  17.02
2         Datsun 710    3.85  2.320  18.61
3     Hornet 4 Drive    3.08  3.215  19.44
4  Hornet Sportabout    3.15  3.440  17.02

In [38]:
df[['Points','Score','Weigh']].agg(['mean','median','std','min','max','var'])

Points     Score      Weigh
mean    3.596563  3.217250  17.848750
median  3.695000  3.325000  17.710000
std     0.534679  0.978457   1.786943
min     2.760000  1.513000  14.500000
max     4.930000  5.424000  22.900000
var     0.285881  0.957379   3.193166

# explanaton for question 9(a) and 9(b) in  assignment 1

# from scipy.stats import skew and kurtosis

In [39]:
from scipy.stats import skew, kurtosis

In [40]:
df1 = pd.read_csv('Q9_a.csv')

In [41]:
df1.head()

Index  speed  dist
0      1      4     2
1      2      4    10
2      3      7     4
3      4      7    22
4      5      8    16

# calculate skew and kurtosis for speed and distance

In [44]:
skew(df1['speed']), skew(df1['dist'])

(-0.11395477012828319, 0.7824835173114966)

# explanation

# from the above, the speed of a car is mostly left skewed and the distance of a car is mostly right skewed . It is clearly shows that the speed and distance are inversly proportional

In [45]:
kurtosis(df1['speed']), skew(df1['dist'])

(-0.5771474239437371, 0.7824835173114966)

# explanation

# from the above , the curve of speed of a car is flatly distributed and the curve of distance of speed of the car is sharply distributed, means the distance values are not far lies from thier mean , where as the speed values of a car is far lies from their mean of the given data

In [46]:
df2 = pd.read_csv('Q9_b.csv')

In [47]:
df2.head()

Unnamed: 0          SP         WT
0           1  104.185353  28.762059
1           2  105.461264  30.466833
2           3  105.461264  30.193597
3           4  113.461264  30.632114
4           5  104.461264  29.889149

In [49]:
skew(df2['SP']),skew(df2['WT'])

(1.5814536794423764, -0.6033099322115126)

# explanation

# from the above , we can say that the SP value of the data is mostly positively skewed and weight of the data is mostly negatively skewed

In [50]:
kurtosis(df2['SP']),skew(df2['WT'])

(2.7235214865269244, -0.6033099322115126)

# explanation

# from the above , we can say that the curve of the SP value of the data is sharp and the vales are lies narrowly from their mean, where as the  WT values are flatly curved , means the values are spread in more form their mean.